packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -2421,17 +2425,48 @@ export function detachOffscreenInstance(instance: OffscreenInstance): void {
     );
   }
 
+  const _detachOffscreen = () => {
+    instance._visibility |= OffscreenDetached;
+    let node = currentOffscreenFiber.child;
+    while (node != null) {
+      disappearLayoutEffects(node);
+      disconnectPassiveEffect(node);
+      node = node.sibling;
+    }
+  };
+
   const executionContext = getExecutionContext();
   if ((executionContext & (RenderContext | CommitContext)) !== NoContext) {
-    scheduleMicrotask(() => {
-      instance._visibility |= OffscreenDetached;
-      disappearLayoutEffects(currentOffscreenFiber);
-      disconnectPassiveEffect(currentOffscreenFiber);
```

Comment:
This is to avoid unmounting refs on offscreen in detach. Instead I unmount effects on all children.